src/app/helpers/custom-errors/baw-api-error.ts

```diff
@@ -62,17 +69,16 @@ export interface ApiErrorDetails {
 /**
  * Determine if error response has already been processed
  *
- * @param errorResponse Error response
+ * @param error Error response
  */
-export function isApiErrorDetails(
-  errorResponse: any
-): errorResponse is ApiErrorDetails {
-  if (!errorResponse) {
+export function isApiErrorDetails(error: any): error is ApiErrorDetails {
+  if (!error || !(error instanceof Object)) {
     return false;
   }
 
-  const keys = Object.keys(errorResponse);
+  const keys = Object.keys(error);
+  keys.sort();
   return (
-    keys.length <= 3 && keys.includes("status") && keys.includes("message")
+    keys === ["message", "status"] || keys === ["info", "message", "status"]
```

Comment:
I've read this a few times and I don't really understand the logic.

If you used `includes` you wouldn't have to sort the keys and this would also be forwards compatible with additional fields added to the error info object.

Maybe add some docs here for future readers?